### PR TITLE
Integrate Date Picker into material_gallery

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: examples/material_gallery/lib/main.dart [hunk 1]
```diff
@@ -6,16 +6,20 @@ import 'package:flutter/material.dart';
 
 import 'chip_demo.dart';
 import 'gallery_page.dart';
+import 'picker_demo.dart';
 import 'widget_demo.dart';
 
 final List<WidgetDemo> _kDemos = <WidgetDemo>[
-  kChipDemo
+  kChipDemo,
+  kPickerDemo,
```
Comment (HansMuller) [inline examples/material_gallery/lib/main.dart]: Shouldn't this be kDatePicker here and similarly elsewhere?

Comment (abarth) [inline examples/material_gallery/lib/main.dart]: I was going off the URLs in the material design spec:

https://www.google.com/design/spec/components/pickers.html

but we can choose better names.

File: examples/material_gallery/lib/picker_demo.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class PickerDemo extends StatefulComponent {
+  _PickerDemoState createState() => new _PickerDemoState();
+}
+
+class _PickerDemoState extends State<PickerDemo> {
+  DateTime _selectedDate = new DateTime.now();
+
+  Future _handleSelectDate() async {
+    _selectedDate = await showDatePicker(
```
Comment (HansMuller) [inline examples/material_gallery/lib/picker_demo.dart]: Should there be a setState() here?

Comment (abarth) [inline examples/material_gallery/lib/picker_demo.dart]: Yes, definitely.

File: examples/material_gallery/lib/main.dart (hunk 1 above)
Comment (HansMuller) [inline examples/material_gallery/lib/main.dart]: I would think we'd have a separate DatePicker and a TimePicker, and corresponding demos.

**Comment by HansMuller** [comment]
LGTM

File: examples/material_gallery/lib/main.dart (hunk 1 above)
Comment (abarth) [inline examples/material_gallery/lib/main.dart]: Done.

**Pull request merged**